### PR TITLE
Remove all `import Foundation` and replace with specific imports.

### PR DESCRIPTION
**Pull request by fabianfett**
NOTE: This can be merged into master!

This will make much clearer where we depend on `Foundation`. In the future we can start to get rid of those parts.

### CODE REVIEW

File: Sources/AWSSDKSwiftCore/String.swift [hunk 1]
```diff
@@ -6,7 +6,42 @@
...
+    "nil",
+    "in",
+    "import",
+    "as",
+    "is",
+    "do",
+    "try",
+    "type",
+    "repeat"
+]
```
Comment (adam-fowler) [inline Sources/AWSSDKSwiftCore/String.swift]: and why is this here?

Comment (fabianfett) [reply Sources/AWSSDKSwiftCore/String.swift]: cherry pick. damn it!

Comment (fabianfett) [reply Sources/AWSSDKSwiftCore/String.swift]: fixed.

Comment (adam-fowler) [reply Sources/AWSSDKSwiftCore/String.swift]: I'm not a keen fan of cherry pick for this very reason
